Dockerfile

```diff
@@ -1,25 +1,21 @@
-FROM node:10.14.1
+FROM node:10.14.2-alpine as builder
 
-RUN echo "deb http://ftp.us.debian.org/debian unstable main contrib non-free" >> /etc/apt/sources.list.d/unstable.list
+RUN mkdir -p /usr/src/app
+WORKDIR /usr/src/app
 
-RUN apt-get update && apt-get install -y rsync \
-                       gcc-5 \
-                       g++-5 \
-                       && rm -rf /var/lib/apt/lists/* \
-                       && rm /etc/apt/sources.list.d/unstable.list
+COPY package*.json /usr/src/app/
+COPY . /usr/src/app/
 
-RUN rm /usr/bin/g++ && ln -s /usr/bin/g++-5 /usr/bin/g++
+RUN apk add --no-cache --virtual .gyp python make git g++ libtool autoconf automake rsync \
+    && npm install
```

Comment:
Curious about this:
- Why running `npm install` like this rather than in a separate line?
- Why no longer `npm ci`?

Reply:
running the installs in one directive reduces the digest layers, making it easier to make a smaller final Docker image. Since the Docker image is a mutli stage build, it makes it a bit simpler. I can move the npm install to its on `RUN` directive and see the outcomes.

`npm ci` is not available in the node:alpine images. I did try get it working but ran into some issues. Let me spend some more time on it this morning

Reply:
Got it! I'm fine with running installs in a single directive.

Would prefer `npm ci` over `npm i` though. I think `npm i` may yield different results for the same package.json and package-lock.json, so we may wind up with very slight (but possible annoying or even problematic) differences. But as long as it's working, I think we're fine merging it as it is and figuring it out in a separate PR.

I'm curious though, why isn't it available in alpine images? It comes with `npm`. It is using an old version of npm?

Reply:
yeah I was stumped as well... oh wait, I think I may have done something wrong previously, npm ci is working now ! :) one second... will push the updates now